### PR TITLE
Remove if

### PR DESCRIPTION
**Pull request by tissue3**
To fix 1) of #70

### CODE REVIEW

File: calyx/src/passes/remove_if.rs [hunk 1]
```diff
@@ -88,12 +88,18 @@ impl Visitor for RemoveIf {
...
-                comps.append(&mut fbranch.comps.clone());
 
-                Ok(Action::Change(ast::Control::enable(comps)))
+                let mut comps_seq = vec![];
+                let mut comps1 = vec![];
+                let mut comps2 = vec![];
+                comps1.append(&mut con.cond);
+                comps2.append(&mut tbranch.comps.clone());
+                comps2.append(&mut fbranch.comps.clone());
+                comps_seq.push(ast::Control::enable(comps1) );
```
Comment (rachitnigam) [inline calyx/src/passes/remove_if.rs]: Is there a reason to create a new vector `comps1`, append all the elements in `con.cond` into and then pass it to `comps_seq.push(enable(comps1)`.

Can't you do `comps_seq.push(enable(con.cond.clone()))`?

File: calyx/src/passes/remove_if.rs [hunk 2]
```diff
@@ -88,12 +88,18 @@ impl Visitor for RemoveIf {
...
-                comps.append(&mut tbranch.comps.clone());
-                comps.append(&mut fbranch.comps.clone());
 
-                Ok(Action::Change(ast::Control::enable(comps)))
+                let mut comps_seq = vec![];
+                let mut comps1 = vec![];
+                let mut comps2 = vec![];
+                comps1.append(&mut con.cond);
+                comps2.append(&mut tbranch.comps.clone());
+                comps2.append(&mut fbranch.comps.clone());
```
Comment (rachitnigam) [inline calyx/src/passes/remove_if.rs]: A different way of creating `comps2` is this:
```rust
                let branch_control: Vec<String> = tbranch.comps
                    .clone()
                    .into_iter()
                    .chain(fbranch.comps.clone().into_iter())
                    .collect();
```
this avoids creating an intermediate vector like `comps2` (as far as a I can tell).

File: calyx/src/passes/remove_if.rs (hunk 1 above)
Comment (tissue3) [reply calyx/src/passes/remove_if.rs]: updated

File: calyx/src/passes/remove_if.rs (hunk 2 above)
Comment (tissue3) [reply calyx/src/passes/remove_if.rs]: updated

**Comment by rachitnigam** [comment]
LGTM, merge!

**Pull request merged**